lib/cc/analyzer/path_patterns.rb

```diff
@@ -10,30 +10,40 @@ def expanded
         @expanded ||= expand
       end
 
+      def match?(path)
+        @patterns.any? do |pattern|
+          value = glob_value(pattern)
+          File.fnmatch(pattern, path)
```

Comment:
OK, I think I'm tracking better now...

Assuming `@patterns` are coming in as `cc/yaml` `Glob` instances, then you're right in that my point number 1 above is being followed (thanks to `#glob_value`). You should only ever be seeing `**/*.rb` here.

However, point number 2 is not. By using `File.fnmatch?` you're going to return `false` here for `path = "foo.rb"`. I suggest you use `PathPatterns`.